blas/src/KokkosBlas1_axpby.hpp

```diff
@@ -21,160 +21,273 @@
+/// \tparam XMV 1-D Kokkos::View or 2-D Kokkos::View. It must
+///             have the same rank as YMV.
+/// \tparam BV  Scalar or 0-D Kokkos::View or 1-D Kokkos::View.
+/// \tparam YMV 1-D Kokkos::View or 2-D Kokkos::View.
 ///
-/// \param a [in] view of type AV, scaling parameter for X.
-/// \param X [in] input view of type XMV.
-/// \param b [in] view of type BV, scaling parameter for Y.
-/// \param Y [in/out] view of type YMV in which the results will be stored.
+/// \param a     [in] Input of type AV:
+///                   - scaling parameter for 1-D or 2-D X,
+///                   - scaling parameters for 2-D X.
+/// \param X     [in] View of type XMV. It must have the same
+///                   extent(s) as Y.
+/// \param b     [in] input of type BV:
+///                   - scaling parameter for 1-D or 2-D Y,
+///                   - scaling parameters for 2-D Y.
+/// \param Y [in/out] View of type YMV in which the results will be
+///                   stored.
 template <class AV, class XMV, class BV, class YMV>
 void axpby(const AV& a, const XMV& X, const BV& b, const YMV& Y) {
   axpby(typename XMV::execution_space{}, a, X, b, Y);
 }
 
 /// \brief Computes Y := a*X + Y
 ///
-/// This function is non-blocking and thread-safe
+/// This function is non-blocking and thread-safe.
 ///
-/// \tparam execution_space a Kokkos execution space where the kernel will run.
-/// \tparam AV 1-D or 2-D Kokkos::View specialization.
-/// \tparam XMV 1-D or 2-D Kokkos::View specialization.
-/// \tparam YMV 1-D or 2-D Kokkos::View specialization. It must have
-///   the same rank as XMV.
+/// \tparam execution_space The type of execution space where the kernel
+///                         will run.
+/// \tparam AV              Scalar or 0-D or 1-D Kokkos::View.
+/// \tparam XMV             1-D or 2-D Kokkos::View. It must have the
+///                         the same rank as YMV.
+/// \tparam YMV             1-D or 2-D Kokkos::View.
 ///
-/// \param space [in] the execution space instance on which the kernel will run.
-/// \param a [in] view of type AV, scaling parameter for X.
-/// \param X [in] input view of type XMV.
-/// \param Y [in/out] view of type YMV in which the results will be stored.
+/// \param pExecSpace [in] The execution space instance on which the kernel
+///                        will run.
+/// \param a          [in] Input of type AV:
+///                        - scaling parameter for 1-D or 2-D X,
+///                        - scaling parameters for 2-D X.
+/// \param X          [in] View of type XMV. It must have the same
+///                        extent(s) as Y.
+/// \param Y      [in/out] View of type YMV in which the results will be
+///                        stored.
 template <class execution_space, class AV, class XMV, class YMV>
-void axpy(const execution_space& space, const AV& a, const XMV& X,
+void axpy(const execution_space& pExecSpace, const AV& a, const XMV& X,
```

Comment:
I kept the template parameter 'execution_space' the same. I changed 'pExecSpace' to 'exec_space'. The change on name of the variable does not cause any problem during the build process (changing the name 'execution_space' causes the build to fail, though). I prefer 'exec_space' to 'space' because it makes it easier to search for the variable. Minor benefit.